### DISCUSSION TITLE
Disable all camera animations - Question

### DISCUSSION THREAD

**Discussion by ahmadz-qmic** [maplibre/maplibre-gl-js#3081] [Q&A]
Hi all,

I was looking all over in both maplibre and mapbox and I couldn't find a way to disable all camera animations. What I mean by that is that I want to have any zoom action mimic a jumpTo without any animations or delays.

Is it possible to do that? If so, I want to disable any transitions without affecting any scrollZoom or panning capabilities (e.g. if I scroll up or down, then the final zoom is set immediately instead of animating) etc...

Any info would be much appreciated.

Thank you.

**Comment by HarelM** [comment]
You can use `prefers-reduced-motion` css for that.
Also see the following example:
https://maplibre.org/maplibre-gl-js/docs/examples/camera-animation/

**Comment by ahmadz-qmic** [comment]
@HarelM thank you for your reply.

However, `prefers-reduced-motion` is only honored if the user settings in the OS allow for reduced motion. 
And as for the example you provided, I know we can control the camera option for `flyTo`, but what about `scrollZoom`? Or any zooming action.

I want to be able to toggle dynamically zoom behavior to mimic a `jumpTo` whenever a user does a zoom action.

**Comment by ahmadz-qmic** [comment]
@HarelM I would love to get your view on this and the feasibility of such implementation.

Thank you!

**Comment by HarelM** [comment]
I'm not sure it's an OS settings, but I haven't tried it myself.
If the only part that is missing here is the zoom handler, then adding an options flag to disable animation should be relatively easy.
See the following code:
https://github.com/maplibre/maplibre-gl-js/blob/18dcac916ccc0b7e7f1160533e5db5c34c28fffe/src/util/browser.ts#L49

And also see #3068
Feel free to investigate it and open a PR.
